pkg/kubelet/apis/device-plugin/v1alpha1/api.proto

```diff
@@ -0,0 +1,90 @@
+// To regenerate api.pb.go run hack/update-generated-runtime.sh
+syntax = 'proto3';
+
+package deviceplugin;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+option (gogoproto.goproto_stringer_all) = false;
+option (gogoproto.stringer_all) =  true;
+option (gogoproto.goproto_getters_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_unrecognized_all) = false;
+
+
+service Registration {
+	rpc Register(RegisterRequest) returns (RegisterResponse) {}
+}
+
+service DevicePlugin {
+	rpc ListWatch(Empty) returns (stream Device) {}
+	rpc Allocate(AllocateRequest) returns (AllocateResponse) {}
+}
+
+message RegisterRequest {
+	// Version of the API the Device Plugin was built against
+	string version = 1;
+	// Name of the unix socket the device plugin is listening on
+	string unixsocket = 2;
+}
+
+message RegisterResponse {
+	// Minimum version the Kubelet API supports.
+	string version = 1;
+	// Kubelet fills this field if it encounters any errors
+	// during the registration process or discover process
+	string error = 2;
+}
+
+message AllocateRequest {
+	repeated Device devices = 1;
+}
+
+message AllocateResponse {
+	repeated DeviceSpec spec = 1;
+}
+
+message DeviceSpec {
+	// List of environment variable to set in the container.
+	repeated KeyValue envs = 1;
+	// Mounts for the container.
+	repeated Mount mounts = 2;
+}
+
+message Mount {
+	string name = 1;
+	string host_path = 2;
+	string mount_path = 3;
+	bool read_only = 4;
+}
+
+message KeyValue {
+	string key = 1;
+	string value = 2;
+}
+
+/* E.g:
+* struct Device {
+*    Kind: "nvidia-gpu"
+*    Name: "GPU-fef8089b-4820-abfc-e83e-94318197576e"
+*    Properties: {
+*        "Family": "Pascal",
+*        "Memory": "4G",
+*        "ECC"   : "True",
+*    }
+*}
+*/
+message Device {
+	string kind = 1;
+	string name = 2;
+	string state = 3;
+
+	map<string, string> properties = 4; // Could be [1, 1.2, 1G]
+
+	DeviceSpec spec = 5;
```

Comment:
I was thinking that we could make Allocate optional if it is the only API exported from device plugin service. But given that we have decided to export ListDevice from device plugin service, there won't be an easy way to make Allocate optional. I am now fine with keeping DeviceRuntimeInfo in Allocate response and skip that from ListDevice.